controller/logout.go

```diff
@@ -33,27 +31,37 @@ func NewLogoutController(service *goa.Service, app application.Application) *Log
 
 // Logout runs the logout action.
 func (c *LogoutController) Logout(ctx *app.LogoutLogoutContext) error {
-	return c.doLogout(ctx, ctx.Redirect, ctx.Referer, ctx.ResponseData, nil)
+	logoutRedirect, err := c.doLogout(ctx, ctx.Redirect, ctx.Referer)
+	if err != nil {
+		log.Error(ctx, nil, "Failed to logout.")
+		return jsonapi.JSONErrorResponse(ctx, err)
```

Comment:
This service is also used by the old logout endpoint, right? So, if the token is nil then just set tokenString to “” otherwise we break the current API. If I’m not missing anything :)
Goa will validate the token in the request for the new endpoint for us.